pypdf/_writer.py

```diff
@@ -148,53 +149,70 @@ class PdfWriter(PdfDocCommon):
     """
 
     def __init__(
         self,
         fileobj: Union[None, PdfReader, StrByteType, Path] = "",
         clone_from: Union[None, PdfReader, StrByteType, Path] = None,
+        incremental: bool = False,
     ) -> None:
-        self._header = b"%PDF-1.3"
+        self.incremental = incremental
+        """
+        Returns if the PdfWriter object has been started in incremental mode
+        """
+
         self._objects: List[Optional[PdfObject]] = []
-        """The indirect objects in the PDF."""
+        """
+        The indirect objects in the PDF.
+        for the incremental it will be filled with None
+        in clone_reader_document_root
+        """
 
-        """Maps hash values of indirect objects to the list of IndirectObjects.
-           This is used for compression.
+        self._original_hash: List[int] = []
+        """
+        list of hashes after import; used to identify changes
         """
+
         self._idnum_hash: Dict[bytes, Tuple[IndirectObject, List[IndirectObject]]] = {}
+        """
+        Maps hash values of indirect objects to the list of IndirectObjects.
+        This is used for compression.
+        """
 
+        self._id_translated: Dict[int, Dict[int, int]] = {}
         """List of already translated IDs.
            dict[id(pdf)][(idnum, generation)]
         """
-        self._id_translated: Dict[int, Dict[int, int]] = {}
-
-        # The root of our page tree node.
-        pages = DictionaryObject()
-        pages.update(
-            {
-                NameObject(PA.TYPE): NameObject("/Pages"),
-                NameObject(PA.COUNT): NumberObject(0),
-                NameObject(PA.KIDS): ArrayObject(),
-            }
-        )
-        self._pages = self._add_object(pages)
-        self.flattened_pages = []
-
-        # info object
-        info = DictionaryObject()
-        info.update({NameObject("/Producer"): create_string_object("pypdf")})
-        self._info_obj: PdfObject = self._add_object(info)
 
-        # root object
-        self._root_object = DictionaryObject()
-        self._root_object.update(
-            {
-                NameObject(PA.TYPE): NameObject(CO.CATALOG),
-                NameObject(CO.PAGES): self._pages,
-            }
-        )
-        self._root = self._add_object(self._root_object)
+        self._ID: Union[ArrayObject, None] = None
+        self._info_obj: PdfObject
+
+        if self.incremental:
+            if isinstance(fileobj, (str, Path)):
+                with open(fileobj, "rb") as f:
+                    fileobj = BytesIO(f.read(-1))
+            if isinstance(fileobj, BytesIO):
+                fileobj = PdfReader(fileobj)
+            else:
+                raise PyPdfError("Invalid type for incremental mode")
+            self._reader = fileobj  # prev content is in _reader.stream
+            self._header = fileobj.pdf_header.encode()
+            self._readonly = True  # !!!TODO: to be analysed
```

Comment:
Is there something to do here?

Reply:
I keep the TODO until the PR I'm currently working on dealing with handling of pages.